### ISSUE TITLE
CA1710 Clearing default suffixes doesn't work

### ISSUE THREAD

**Issue by erichiller** [dotnet/roslyn-analyzers#5035]
### Analyzer

**Diagnostic ID**: [CA1710](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1710): `Identifiers should have correct suffix`
### Analyzer source

Versions:
```
.NET SDK (reflecting any global.json):
 Version:   5.0.300-preview.21180.15
 Commit:    19ce2535ee

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.300-preview.21180.15\

Host (useful for support):
  Version: 5.0.4
  Commit:  f27d337295

.NET SDKs installed:
  3.1.102 [C:\Program Files\dotnet\sdk]
  3.1.302 [C:\Program Files\dotnet\sdk]
  3.1.401 [C:\Program Files\dotnet\sdk]
  5.0.100 [C:\Program Files\dotnet\sdk]
  5.0.200-preview.20601.7 [C:\Program Files\dotnet\sdk]
  5.0.300-preview.21180.15 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

I also tried installing the Nuget package, this did not change anything.

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

Overriding, or disabling built-in suffixes as shown at the bottom of the [CA1710 Docs page](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1710#additional-required-suffixes), ( see `dotnet_code_quality.CA1710.additional_required_suffixes = T:System.Data.IDataReader->{}` ) does not disable the warnings, at least not for `IReadOnlyDictionary`

Here are the settings I've used in `.editorconfig`, note the `dotnet_code_quality.ca1710.additional_required_suffixes` line which follows the format found in the docs:

```EditorConfig
dotnet_diagnostic.ca1710.severity = warning
dotnet_code_quality.ca1710.additional_required_suffixes = T:System.Collections.Generic.IReadOnlyDictionary`2->{}
```
Note that the same errors are present using the settings (sans the `` `2 `` suffix of `IReadOnlyDictionary` )
```EditorConfig
dotnet_code_quality.ca1710.additional_required_suffixes = T:System.Collections.Generic.IReadOnlyDictionary->{}
```

Nor does using the alternate plaintext version of the identifier:
```EditorConfig
dotnet_code_quality.ca1710.additional_required_suffixes = IReadOnlyDictionary`2->{}
```
or
```EditorConfig
dotnet_code_quality.ca1710.additional_required_suffixes = IReadOnlyDictionary->{}
```

I know the settings are being read, because using the following setting will make the error go away ( as long as the suffix is `s` ):
```EditorConfig
dotnet_code_quality.ca1710.additional_required_suffixes = T:System.Collections.Generic.IReadOnlyDictionary`2->s
```

Here is the error shown in VSCode and Visual Studio, both are on the latest versions.

In VSCode (version 1.55.2, C# extension version: 1.23.11)
![image](https://user-images.githubusercontent.com/5379325/115227664-fc813c80-a0cd-11eb-95ae-9eea659d9eb0.png)

In Visual Studio (2019 Preview, Version 16.10.0 Preview 2.0)
![image](https://user-images.githubusercontent.com/5379325/115229154-f0967a00-a0cf-11eb-8d7f-77f2d42c0d24.png)

Additionally, to be thorough I also tried as types of non abstract `record` as well as `class` and `interface`

### Steps To Reproduce

Insert the override setting into the project's `.editorconfig`:
```
dotnet_diagnostic.ca1710.severity = warning
dotnet_code_quality.ca1710.additional_required_suffixes = T:System.Collections.Generic.IReadOnlyDictionary`2->{}
```

Create a record or other user defined type that implements `IReadOnlyDictionary`, observe the error, eg
```cs
    public abstract record IndicatorParameters : IReadOnlyDictionary<string, object> {
        /// <summary>
        /// Indicator Parameters as an <see cref="IReadOnlyDictionary{TKey,TValue}"/> to be returned
        /// </summary>
        protected abstract IReadOnlyDictionary<string, object> _dict { get; }

        /// <inheritdoc />
        public IEnumerator<KeyValuePair<string, object>> GetEnumerator( ) => this._dict.GetEnumerator();

        /// <inheritdoc />
        IEnumerator IEnumerable.GetEnumerator( ) => ( ( IEnumerable )this._dict ).GetEnumerator();

        /// <inheritdoc />
        public int Count => this._dict.Count;

        /// <inheritdoc />
        public bool ContainsKey( string key ) => this._dict.ContainsKey( key );

        /// <inheritdoc />
        public bool TryGetValue( string key, out object value ) => this._dict.TryGetValue( key, out value );

        /// <inheritdoc />
        public object this[ string key ] => this._dict[ key ];

        /// <inheritdoc />
        public IEnumerable<string> Keys => this._dict.Keys;

        /// <inheritdoc />
        public IEnumerable<object> Values => this._dict.Values;
    }
```

### Expected behavior

Upon setting up the override for `IReadOnlyDictionary` there would no longer be a `CA1710` error.

### Actual behavior

There is an error on implementors of `IReadOnlyDictionary` despite the aforementioned override configured in `.editorconfig`
